module/VuFindSearch/src/VuFindSearch/Backend/AbstractBackend.php

```diff
@@ -116,6 +126,39 @@ abstract public function getRecordCollectionFactory();
     protected function injectSourceIdentifier(RecordCollectionInterface $response)
     {
         $response->setSourceIdentifiers($this->identifier);
+        $response->setResultSetIdentifier($this->generateUuid());
+
         return $response;
     }
+
+    /**
+     * Sets the result set identifier for the record collection.
+     *
+     * @param string $uuid A valid UUID associated with the data set.
+     *
+     * @return void
+     */
+    public function setResultSetIdentifier(string $uuid)
+    {
+        $this->recordCollection->setResultSetIdentifier($uuid);
```

Comment:
Is this method actually used? I think the property you were forced to add was related to the call in this method, but I suspect this method is unnecessary.